VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -54,27 +66,48 @@ public void call(Session session, SessionState state, Exception exception) {
         }
     };
 
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    ProfileModel profile;
+
+    ImageView imageView;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         uiLifecycleHelper = IUiLifecycleHelper.Factory.getInstance(getActivity(), callback);
         uiLifecycleHelper.onCreate(savedInstanceState);
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState) {
-        super.onCreateView(inflater, container, savedInstanceState);
         Context context = getActivity().getBaseContext();
-
         socialPref = new PrefManager(context, PrefManager.Pref.FEATURES);
-
         pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+        profile = pref.getCurrentUserProfile();
+        if (config.isUserProfilesEnabled() && profile != null && profile.username != null) {
+            getAccountTask = new GetAccountTask(getActivity(), profile.username) {
+                @Override
+                protected void onSuccess(Account account) throws Exception {
+                    final ProfileImage profileImage = account.getProfileImage();
+                    Glide.with(NavigationFragment.this)
+                            .load(profileImage.getImageUrlLarge())
```

Comment:
Is this what you mean?
https://github.com/edx/edx-app-android/commit/13f195b313e7e01f2c8b04aee7280070614ef492

On Mon, Sep 28, 2015 at 9:16 AM, Ibraheem Zaman <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java
> <https://github.com/edx/edx-app-android/pull/372#discussion_r40549958>:
>
> >          pref = new PrefManager(context, PrefManager.Pref.LOGIN);
> > +        profile = pref.getCurrentUserProfile();
> > +        if (config.isUserProfilesEnabled() && profile != null && profile.username != null) {
> > +            getAccountTask = new GetAccountTask(getActivity(), profile.username) {
> > +                @Override
> > +                protected void onSuccess(Account account) throws Exception {
> > +                    final ProfileImage profileImage = account.getProfileImage();
> > +                    Glide.with(NavigationFragment.this)
> > +                            .load(profileImage.getImageUrlLarge())
>
> Perhaps this should be loaded as a thumbnail, while the larger version
> would be the full-sized resource.
> … <#150141881d242265_>
> On Fri, Sep 25, 2015 at 8:28 PM, Brian Guertin <notifications@github.com>
> wrote: In
> VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java <
> https://github.com/edx/edx-app-android/pull/372#discussion_r40442068>: >
> pref = new PrefManager(context, PrefManager.Pref.LOGIN); > + profile =
> pref.getCurrentUserProfile(); > + if (config.isUserProfilesEnabled() &&
> profile != null && profile.username != null) { > + getAccountTask = new
> GetAccountTask(getActivity(), profile.username) { > + @Override
> <https://github.com/Override> > + protected void onSuccess(Account
> account) throws Exception { > + final ProfileImage profileImage =
> account.getProfileImage(); > + Glide.with(NavigationFragment.this) > +
> .load(profileImage.getImageUrlLarge()) Despite the name, large is actually
> not very large, only 120px wide. So in fact, it is not really large enough,
> since we need 140px@2x, 210px@3x, and 280px@4x. Unfortunately the next
> size up is 500px, but maybe we should just load that one. — Reply to this
> email directly or view it on GitHub <
> https://github.com/edx/edx-app-android/pull/372/files#r40442068>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/372/files#r40549958>.
>